taskvine/src/manager/vine_manager.c

```diff
@@ -380,6 +418,11 @@ static int handle_cache_update(struct vine_manager *q, struct vine_worker_info *
 		vine_current_transfers_remove(q, id);
 
 		vine_txn_log_write_cache_update(q, w, size, transfer_time, start_time, cachename);
+
+		// If the file was not a url, eg. 'X', and it is a temp
+		if (*id == 'X' && strncmp(cachename, "temp-rnd-", 9) == 0) {
```

Comment:
This seems unavoidable for now, but I would rather have some direct information present in the `vine_file_replica` structure that tells you exactly what kind it is, whether it is a temporary file, etc, rather than inferring it from the name.

To make this work, the manager would have to send the relevant data to the worker, so that it could be returned back after failure recovery.